Sources/Boundaries/Reducer.swift

```diff
@@ -11,45 +11,50 @@ public struct Prism<A, B> {
   }
 }
 
-public struct Reducer<S, A> {
-  public let reduce: (A, S) -> (S, Effect<A>)
+public struct Reducer<S, A, E: EffectProtocol> {
+  public let reduce: (A, S) -> (S, Cmd<E>)
 
-  public init(_ reduce: @escaping (A, S) -> (S, Effect<A>)) {
+  public init(_ reduce: @escaping (A, S) -> (S, Cmd<E>)) {
     self.reduce = reduce
   }
 
-  public func lift<B>(action prism: Prism<B, A>) -> Reducer<S, B> {
+  public func lift<B>(action prism: Prism<B, A>) -> Reducer<S, B, E> {
     return .init { actionB, state in
       prism
         .preview(actionB)
-        .map { self.reduce($0, state) |> second(map(prism.review)) }
```

Comment:
this `second(map(prism.review))` simply embedded the `Cmd`'s sub-action of type `A` into the super-action `B`, and then wrapped it into another `Cmd`. This means reducers could never run effects that dispatch actions it doesn't understand. That seems needlessly restrictive.

Reply:
Still wondering how these exceptions'll look in practice.